File: 0494-target-sum/0494-target-sum.java
class Solution {
    int cnt = 0;
    public void fun(int [] nums, int sum, int target, int ind){
        if(ind == nums.length){
            if(sum == target){
                cnt++;
            }
            return;
        }
        fun(nums, sum+nums[ind], target, ind+1);
        fun(nums, sum-nums[ind], target, ind+1);
    }
    public int findTargetSumWays(int[] nums, int target) {
        fun(nums, 0, target, 0);
        return cnt;
    }
}

File: 0494-target-sum/0494-target-sum.cpp
class Solution {
public:
    int cnt = 0;
    void fun(vector<int>&nums, int target, int ind, int sum){
        if(ind == nums.size()){
            if(sum == target){
                cnt++;
            }
            return;
        }
        fun(nums, target, ind+1, sum-nums[ind]);
        fun(nums, target, ind+1, sum+nums[ind]);
    }
    int findTargetSumWays(vector<int>& nums, int target) {
        fun(nums, target, 0, 0);
        return cnt;
    }
};